product/domain/productBasics.go

```diff
@@ -570,13 +580,30 @@ func findMediaInProduct(p BasicProduct, group string, usage string) *Media {
 
 // IsInStock returns information if current product whether in stock or not
 func (bpd BasicProductData) IsInStock() bool {
+	for _, stock := range bpd.Stock {
+		if stock.InStock {
+			return true
+		}
+	}
+
 	if bpd.StockLevel == "" || bpd.StockLevel == StockLevelOutOfStock {
 		return false
 	}
 
 	return true
 }
 
+// IsInStockForDeliveryCode returns information if current product whether in stock or not for provided delivery code
+func (bpd BasicProductData) IsInStockForDeliveryCode(deliveryCode string) bool {
+	for _, stock := range bpd.Stock {
+		if stock.DeliveryCode == deliveryCode {
+			return stock.InStock
+		}
+	}
+
+	return true
```

Comment:
I would say out of stock when we do not have info for the deliveryCode
```suggestion
	return false
```